src/main/java/tel/schich/javacan/BcmMessage.java

```diff
@@ -0,0 +1,308 @@
+        return (sec + usec) != 0 ? new BcmTimeval(sec, usec) : null;
+    }
+
+    /**
+     * Returns the CAN ID of this message.
+     */
+    public int getCanId() {
+        return buffer.getInt(base + OFFSET_CAN_ID);
+    }
+
+    /**
+     * Returns the number of frames in this message.
+     */
+    public int getNFrames() {
+        return buffer.getInt(base + OFFSET_NFRAMES);
+    }
+
+    /**
+     * Returns a single frame of this message.
+     *
+     * @param index of the frame; ({@code 0 <= index < nFrames})
+     * @return the frame
+     * @throws IllegalArgumentException if the message buffer contains no frame for that index
+     */
+    public CanFrame getFrame(int index) {
+        ByteBuffer frameBuffer = buffer.duplicate();
+        int frameLength = frameLength(getFlags());
+        frameBuffer
+                .position(base + OFFSET_FRAMES + index * frameLength)
+                .limit(frameBuffer.position() + frameLength);
+        return CanFrame.create(frameBuffer);
+    }
+
+    /**
+     * Returns all frames of this message.
+     *
+     * @return all frames of this message
+     */
+    public List<CanFrame> getFrames() {
+        int nFrames = getNFrames();
+        if (nFrames == 0) {
+            return Collections.emptyList();
+        }
+        int frameLength = frameLength(getFlags());
+        List<CanFrame> frames = new ArrayList<>(nFrames);
+        for (int i = 0; i < nFrames; i++) {
+            ByteBuffer frameBuffer = buffer.duplicate();
+            frameBuffer
+                    .position(base + OFFSET_FRAMES + i * frameLength)
+                    .limit(frameBuffer.position() + frameLength);
+            frames.add(CanFrame.create(frameBuffer));
+
+        }
+        return frames;
+    }
+
+    /**
+     * Returns the backing {@link ByteBuffer} with proper position and limit set to read the entire BCM
+     * message.
+     *
+     * @return the backing buffer
+     */
+    public ByteBuffer getBuffer() {
+        this.buffer.clear().position(base).limit(base + size);
+        return this.buffer;
+    }
+
+    private static int frameLength(Set<BcmFlag> flags) {
+        return flags.contains(BcmFlag.CAN_FD_FRAME) ? RawCanChannel.FD_MTU : RawCanChannel.MTU;
+    }
+
+    private long getPlatformLong(int offset) {
+        switch (BcmMessage.LONG_SIZE) {
+        case 4:
+            return buffer.getInt(offset);
+        case 8:
+            return buffer.getLong(offset);
+        default:
+            throw new UnsupportedPlatformException();
+        }
+    }
+
+    private void putPlatformLong(int offset, long value) {
+        switch (BcmMessage.LONG_SIZE) {
+        case 4:
+            buffer.putInt(offset, (int) value);
+            break;
+        case 8:
+            buffer.putLong(offset, value);
+            break;
+        default:
+            throw new UnsupportedPlatformException();
+        }
+    }
+
+    private static native int getLongSize();
```

Comment:
I think these platform link things should be moved to a separate helper, they are not directly related to BCM.

Reply:
If the library accumulates more than this one method i tend to agree, but currently a helper just introduces an additional indirection.

Reply:
On second thought I agree. I originally anticipated possibly using it for the raw frames, but completely forgot they were fixed anyway.